shell/browser/web_contents_permission_helper.cc

```diff
@@ -157,10 +156,15 @@ void WebContentsPermissionHelper::RequestWebNotificationPermission(
 }
 
 void WebContentsPermissionHelper::RequestPointerLockPermission(
-    bool user_gesture) {
+    bool user_gesture,
+    bool last_unlocked_by_target,
+    base::OnceCallback<void(content::WebContents*, bool, bool, bool)>
+        callback) {
   RequestPermission(
       static_cast<content::PermissionType>(PermissionType::POINTER_LOCK),
-      base::BindOnce(&OnPointerLockResponse, web_contents_), user_gesture);
+      base::BindOnce(&OnPointerLockResponse, std::move(callback), web_contents_,
+                     user_gesture, last_unlocked_by_target),
```

Comment:
```suggestion
      base::BindOnce(std::move(callback), web_contents_,
                     user_gesture, last_unlocked_by_target),
```

Callback can be called without having to redirect to `OnPointerLockResponse`.